### PR TITLE
Revamp progress script, change file structure and remove unused asm

### PR DESCRIPTION
**Pull request by AloXado320**
Improvements and fixes to `progress.py`
- Proper checks for directory and file searching, fixes percentage inaccurracy caused by hardcoded paths.
- Visually pleasing progress report, inspired from DKR decomp, simulating an All Cups run.
- Previous progress view report is now under ./progress.py verbose
- Individual percentages for `main`, `ending`, `racing`, `audio`, `os`, `game` (main + ending + racing) and `total` (all segments).

File sizes were checked from an OK `mk64.us.map` to ensure there is no inflated numbers, spreadsheet for reference:
https://docs.google.com/spreadsheets/d/13dTsQYIbcoQlSKYZVS7yq_GPzp2wQS5gw3FM4P5Z7cs/edit?usp=sharing

Some files were moved around, mostly C data files to clean up directories.

Removed unused non_matching asm files that were already matching, these were affecting percentage accuracy as well.

Readme and Jenkins were also update to account for the individual percentages in `progress.py` and new directories.

### CODE REVIEW

**Review by MegaMech — CHANGES_REQUESTED** [review]

**Comment by MegaMech** [comment]
I'm not going to double check the python script itself because frankly I don't have time.

If I do have time, I want to quickly test the script itself and how it reaches its total bytes number because I was getting a slightly different number. So might be a bit.

**Comment by MegaMech** [comment]
I also still think OS shouldn't be included in the final total. Because it's a library

File: Jenkinsfile [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+def mainProgBadge = addEmbeddableBadgeConfiguration(id: "mainProgress", subject: "Main Code Progress", color: "7d0000")
+def endingProgBadge = addEmbeddableBadgeConfiguration(id: "endingProgress", subject: "Ending Code Progress", color: "7d0000")
+def racingProgBadge = addEmbeddableBadgeConfiguration(id: "racingProgress", subject: "Racing Code Progress", color: "7d0000")
 def audioProgBadge = addEmbeddableBadgeConfiguration(id: "audioProgress", subject: "Audio Code Progress", color: "7d0000")
+def osProgBadge = addEmbeddableBadgeConfiguration(id: "osProgress", subject: "Libultra Code Progress", color: "7d0000")
 
 def bytesToDecompile = addEmbeddableBadgeConfiguration(id: "bytesLeft", subject: "Remaining Decompilable Bytes", color: "7d0000")
-def m2cFuncs = addEmbeddableBadgeConfiguration(id: "m2c", subject: "Remaining Functions", color: "7d0000")
-def nonmatchingFuncs = addEmbeddableBadgeConfiguration(id: "nonmatching", subject: "Non-matching Functions", color: "7d0000")
+def globalAsmFuncs = addEmbeddableBadgeConfiguration(id: "globalasm", subject: "Remaining Functions", color: "7d0000")
```
[the remark this answers is not in the document]
Comment (AloXado320) [reply Jenkinsfile:11]: I could rename it to "remainingFuncs", to sound it less verbose.

File: README.md [hunk 2]
```diff
@@ -78,42 +83,38 @@ Some menu textures are compressed using a format called tkmk00. A byte-matching
...
-	├── asm: Handwritten assembly code, rom header and boot
-	│   ├── non_matchings: asm for non-matching sections
-	│   └── os: OS related assembly code
+	├── asm: Handwritten Assembly code, rom header and boot
+	│   ├── non_matchings: ASM for non-matching sections
+	│   └── os: Libultra handwritten assembly code
 	├── assets: Textures
-	├── bin: Files needing import. Most of them are already done.
-	├── courses: Course geography, course data, course table, and staff ghosts.
+	├── courses: Course data, geography, display lists and staff ghosts
```
[the remark this answers is not in the document]
Comment (AloXado320) [reply README.md:90]: In SM64 decomp read me it's separate and i think that's how it is, the one word might be just for code
https://en.wikipedia.org/wiki/Display_list

[the remark this answers is not in the document]
Comment (MegaMech) [reply README.md:90]: kk

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (AloXado320) [reply Jenkinsfile:11]: For a second i though you were refering to the python script and not Jenkins lol then we are fine

[the remark this answers is not in the document]
Comment (MegaMech) [reply Jenkinsfile:11]: Sorry I meant for the readme.

You're right that wasn't very clear

**Review by MegaMech — APPROVED** [review]

**Pull request merged**